src/WalkingModule/src/Module.cpp

```diff
@@ -1204,14 +1208,32 @@ bool WalkingModule::evaluateZMP(iDynTree::Vector2& zmp)
         return false;
     }
 
+    // rotate the resulting zmp and offset
+    double yawLeft = m_leftTrajectory.front().getRotation().asRPY()(2);
+    double yawRight = m_rightTrajectory.front().getRotation().asRPY()(2);
+
+    // evaluate the mean of the angles
+    double meanYaw = std::atan2(std::sin(yawLeft) + std::sin(yawRight),
+                                std::cos(yawLeft) + std::cos(yawRight));
+    iDynTree::Rotation yawRotation = iDynTree::Rotation::RotZ(meanYaw);
```

Comment:
This is a duplicate of https://github.com/robotology/walking-controllers/blob/2872a321138d53c5cacc519808cee82fe8e75287/src/WalkingModule/src/Module.cpp#L931-L941

We should understand how to avoid this duplication

Reply:
I tried to reduce the duplication in https://github.com/robotology/walking-controllers/pull/137/commits/5019f4093381f9efdae85167cc60b020adfef620